src/classBased/components/TodoItem.js

```diff
@@ -0,0 +1,72 @@
+import React from "react"
+import styles from "./TodoItem.module.css"
+
+class TodoItem extends React.Component {
+  state = {
+    editing: false,
+  }
+
+  handleEditing = () => {
+    this.setState({
+      editing: true,
+    })
+  }
+
+  handleUpdatedDone = event => {
+    if (event.key === "Enter") {
+      this.setState({ editing: false })
+    }
+  }
+
+  componentWillUnmount() {
+    console.log("Cleaning up...")
+  }
+
+  render() {
+    const completedStyle = {
+      fontStyle: "italic",
+      color: "#595959",
+      opacity: 0.4,
+      textDecoration: "line-through",
+    }
+    const { completed, id, title } = this.props.todo
+
+    let viewMode = {}
+    let editMode = {}
+
+    if (this.state.editing) {
+      viewMode.display = "none"
+    } else {
+      editMode.display = "none"
+    }
+
+    return  <li className={styles.item}>
+              <div onDoubleClick={this.handleEditing} style={viewMode}>
+                <input 
+                  type="checkbox" 
+                  className={styles.checkbox}
+                  checked={completed} 
+                  onChange={() => this.props.handleChangeProps(id)}
+                />
+                <button  onClick={() => this.props.deleteTodoProps(id)}>
+                    Delete
```

Comment:
You have to use destructuring for this.props

Reply:
Thanks